### ISSUE TITLE
solveset(x**2 + y, x) for real `x` return Empty Set

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#9557]
```
>>> x = Symbol('x', real=True)
>>> y, z = symbols('y, z')
>>> solveset(x** + y, x)
{}                                 # returns an empty
>>> solve(x**2 + y, x)
[ - sqrt(-y), sqrt(-y) ]
>>> solveset(z**2 + y, z)
{ - sqrt(-y), sqrt(-y) ] 
>>> solveset_complex(x**2 + y, x)
[ - sqrt(-y), sqrt(-y) ]
```

I have not seen code base for `solveset` at all, but this seems not be followed.

**Comment by aktech** [comment]
@gxyd  Thanks for reporting.

This is due to 3rd last line in `solveset_real` :

```python
In [7]: result =  [sqrt(-y), -sqrt(-y)]

In [8]: FiniteSet(*result).intersect(S.Reals)
Out[8]: EmptySet()
```
PR #9540  is supposed to fix this.

**Comment by aktech** [comment]
Closed via #9618
